crates/ironrdp-session/src/x224/mod.rs

```diff
@@ -16,14 +16,26 @@ pub enum ProcessorOutput {
     /// A buffer with encoded data to send to the server.
     ResponseFrame(Vec<u8>),
     /// A graceful disconnect notification. Client should close the connection upon receiving this.
-    Disconnect(DisconnectReason),
+    Disconnect(DisconnectDescription),
     /// Received a [`ironrdp_pdu::rdp::headers::ServerDeactivateAll`] PDU. Client should execute the
     /// [Deactivation-Reactivation Sequence].
     ///
     /// [Deactivation-Reactivation Sequence]: https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/dfc234ce-481a-4674-9a5d-2a7bafb14432
     DeactivateAll(Box<ConnectionActivationSequence>),
 }
 
+#[derive(Debug, Clone)]
+pub enum DisconnectDescription {
+    /// Includes the reason from the MCS Disconnect Provider Ultimatum.
+    /// This is the least-specific disconnect reason and is only used
+    /// when a more specific disconnect code is not available.
+    McsDisconnect(DisconnectReason),
```

Comment:
[suggestion]: Naming - I think `McsDisconnect` is too specific about protocol details, I suggest the following naming. WDYT?:
```rust 
pub enum DisconnectKind {
    Graceful(DisconnectReason), 
    Failure(ErrorInfo),
}
```

Reply:
Ah, right, `Graceful` is not fitting name here, not all `DisconnectReason` variants are graceful disconnects

Reply:
Then I guess it’s good as-is?